himportscan/src/HImportScan/ImportScanner.hs

```diff
@@ -62,31 +62,40 @@ data ModuleImport = ModuleImport
 data ImportMethod = SourceImport | NormalImport
   deriving (Eq, Ord)
 
-instance Aeson.ToJSON ScannedImports where
-  toJSON ScannedImports{..} =
-    Aeson.object $
-      [ ("filePath", Aeson.String filePath)
-      , ("moduleName", Aeson.String moduleName)
-      , ("importedModules", Aeson.toJSON importedModules)
+jsonText :: Text -> Json.JSValue
+jsonText = Json.JSString . Json.toJSString . Text.unpack
+
+instance Json.JSON ScannedImports where
+  showJSON ScannedImports{..} = Json.JSObject $
+    Json.toJSObject $
+      [ ("filePath", jsonText filePath)
+      , ("moduleName", jsonText moduleName)
+      , ("importedModules", Json.showJSON importedModules)
       ] ++
-      [ ("usesTH", Aeson.toJSON True) | usesTH] ++
-      [ ("isBoot", Aeson.toJSON True) | isBoot]
+      [ ("usesTH", Json.JSBool True) | usesTH] ++
+      [ ("isBoot", Json.JSBool True) | isBoot]
+
+  -- We are only exorting the result to JSON, hence the read function is undefined.
+  readJSON _ = undefined
```

Comment:
Very minor nitpick, but I think it's better to have the comment in an `error` instead, so that if this ever occurs, it's easier to trace.

Same for the instance on `ModuleImport`